app/dossierPrograms/dossierPrograms.controllers.js

```diff
@@ -206,6 +206,202 @@ dossierProgramsModule.controller("dossiersProgramSectionController", [
+                        items: [
+                            {
+                                id: "USER_ORGUNIT",
+                            },
+                        ],
+                    },
+                ],
+                rows: [
+                    {
+                        dimension: "dx",
+                        items: [
+                            {
+                                id: "DqqSJFWB392",
+                            },
+                            {
+                                id: "qywsusOdy33",
+                            },
+                        ],
+                    },
+                ],
+                series: [],
+                outlierAnalysis: null,
+                cumulative: false,
+            };
+
+            const sharing = {
+                object: {
+                    id: "LEP0WHTGYUe",
+                    name: "name",
+                    publicAccess: "--------",
+                    externalAccess: false,
+                    userGroupAccesses: [{ id: "epFY01iJN0Z", name: "ALL USERS", access: "rw------" }],
+                },
+            };
+
+            let items = [{ id: id }];
+
+            payload.name = name + " - " + id;
+            payload.rows[0].items = rowItems ? items.concat(rowItems.map(id => ({ id: id }))) : items;
+
+            $scope.table = dossiersProgramVisualizationTableFactory.get_table.query(
+                {
+                    filter: "name:eq:" + payload.name,
+                },
+                function (tbl) {
+                    if (tbl && tbl.visualizations[0]) {
+                        if (tbl.visualizations[0].id) {
+                            console.log("Updating Table");
+                            payload.id = tbl.visualizations[0].id;
+                            sharing.object.id = tbl.visualizations[0].id;
+                            sharing.object.name = tbl.visualizations[0].name;
+
+                            dossiersProgramVisualizationTableFactory.upd_table.query(
+                                {
+                                    uid: tbl.visualizations[0].id,
+                                },
+                                payload,
+                                function (response) {
+                                    dossiersProgramVisualizationTableFactory.upd_sharing.query(
+                                        { uid: tbl.visualizations[0].id },
+                                        sharing,
+                                        function (res) { }
+                                    );
+
+                                    const uid = tbl.visualizations[0].id;
+
+                                    $window.open(dhisroot + "/dhis-web-data-visualizer/index.html#/" + uid, "_blank");
+                                }
+                            );
+                        }
+                    }
+
+                    if (tbl.visualizations[0] == undefined) {
```

Comment:
[minor] is this an `else` of the previous branch?

Reply:
Yes, I overlooked that, done now.